### PR TITLE
fixed chooseSize

### PR DESCRIPTION
**Pull request by don41382**
! @jystic 

fixed implementation of chooseSize, which only returned always the max

### CODE REVIEW

**Comment by charleso** [comment]
@don41382 I wouldn't mind just double-checking this with @jystic, we should wait until he's had a chance to look at this

File: disorder-core/src/Disorder/Core/Gen.hs [hunk 1]
```diff
@@ -37,12 +37,12 @@ import           Prelude
...
 vectorOfSize min' max' gen =
-  chooseSize min' max' >>= flip vectorOf gen
+  choose (min', max') >>= flip vectorOf gen
 
--- | Return an 'Int' which is between the provided range, and influenced by the current "size"
+-- | Return an 'Int' which is between the provided range, and influenced by the current "size" -- deprecated, please use "choose" instead
 chooseSize :: Int -> Int -> Gen Int
 chooseSize min' max' =
-  sized (return . min max' . max min')
+  choose (min', max')
```
Comment (jystic) [inline disorder-core/src/Disorder/Core/Gen.hs]: Given the description I would have expected the implementation to be something like:
```hs
chooseSize :: Int -> Int -> Gen Int
chooseSize x0 y0 =
  sized $ \sz0 ->
    let
      x =
        toInteger x0

      y =
        toInteger y0

      sz =
        toInteger sz0

      diff =
        ((y - x) * sz) `quot` 99
    in
      choose (x0, fromIntegral $ x + diff)
```

Comment (jystic) [reply disorder-core/src/Disorder/Core/Gen.hs]: Actually that probably needs something to clamp `x + diff` so that it doesn't go larger than `y0`

**Comment by charleso** [comment]
@jystic Let's catch up about this when you're back in the office. I honestly can't remember what I was thinking with that function - it's not clear how it's actually useful.

**Comment by jystic** [comment]
I'm here, happy to chat any time :) maybe next tea break?

**Comment by don41382** [comment]
have you already talked about it? Would like to know why we have to do this (crazy) stuff.

**Comment by charleso** [comment]
@don41382 I would take Jacob's code and his comment fix and run with that

**Comment by jystic** [comment]
> Would like to know why we have to do this (crazy) stuff.

Happy to explain how the size parameter works in QuickCheck if you'd like?

**Comment by jystic** [comment]
@don41382 this should be good to go now I think

**Pull request merged**